libs/partners/openai/tests/unit_tests/test_secrets.py

```diff
@@ -60,3 +65,336 @@ def test_azure_openai_embeddings_secrets() -> None:
     s = str(o)
     assert "foo1" not in s
     assert "foo2" not in s
+
+
+# azure chat openai
+def test_azure_chat_openai_api_key_is_secret_string() -> None:
+    """Test that the API key is stored as a SecretStr."""
+    chat_model = AzureChatOpenAI(
+        openai_api_key="secret-api-key",
+        azure_endpoint="endpoint",
+        azure_ad_token="secret-ad-token",
+        api_version="version",
+    )
+    assert isinstance(chat_model.openai_api_key, SecretStr)
+    assert isinstance(chat_model.azure_ad_token, SecretStr)
+
+
+def test_azure_chat_openai_api_key_masked_when_passed_from_env(
+    monkeypatch: MonkeyPatch, capsys: CaptureFixture
+) -> None:
+    """Test that the API key is masked when passed from an environment variable."""
+    monkeypatch.setenv("AZURE_OPENAI_API_KEY", "secret-api-key")
+    monkeypatch.setenv("AZURE_OPENAI_AD_TOKEN", "secret-ad-token")
+    chat_model = AzureChatOpenAI(
+        openai_api_key="secret-api-key",
+        azure_endpoint="endpoint",
+        azure_ad_token="secret-ad-token",
+        api_version="version",
+    )
+    print(chat_model.openai_api_key, end="")  # noqa: T201
+    captured = capsys.readouterr()
+
+    assert captured.out == "**********"
+
+    print(chat_model.azure_ad_token, end="")  # noqa: T201
+    captured = capsys.readouterr()
+
+    assert captured.out == "**********"
+
+
+def test_azure_chat_openai_api_key_masked_when_passed_via_constructor(
+    capsys: CaptureFixture,
+) -> None:
+    """Test that the API key is masked when passed via the constructor."""
+    chat_model = AzureChatOpenAI(
+        openai_api_key="secret-api-key",
+        azure_endpoint="endpoint",
+        azure_ad_token="secret-ad-token",
+        api_version="version",
+    )
+    print(chat_model.openai_api_key, end="")  # noqa: T201
+    captured = capsys.readouterr()
+
+    assert captured.out == "**********"
+
+    print(chat_model.azure_ad_token, end="")  # noqa: T201
+    captured = capsys.readouterr()
+
+    assert captured.out == "**********"
+
+
+def test_azure_chat_openai_uses_actual_secret_value_from_secretstr() -> None:
+    """Test that the actual secret value is correctly retrieved."""
+    chat_model = AzureChatOpenAI(
+        openai_api_key="secret-api-key",
+        azure_endpoint="endpoint",
+        azure_ad_token="secret-ad-token",
+        api_version="version",
+    )
+    assert (
+        cast(SecretStr, chat_model.openai_api_key).get_secret_value()
+        == "secret-api-key"
+    )
+    assert (
+        cast(SecretStr, chat_model.azure_ad_token).get_secret_value()
+        == "secret-ad-token"
+    )
+
+
+# azure openai
+def test_azure_openai_api_key_is_secret_string() -> None:
+    """Test that the API key is stored as a SecretStr."""
+    llm = AzureOpenAI(
```

Comment:
(nit) Are we able to parametrize these at all? e.g.,
```python
from typing import Type

import pytest


@pytest.mark.parametrize(
    "model_class", [AzureChatOpenAI, AzureOpenAI, AzureOpenAIEmbeddings]
)
def test_api_key_is_secret_string(model_class: Type) -> None:
    model_instance = model_class(
        openai_api_key="secret-api-key",
        azure_endpoint="endpoint",
        azure_ad_token="secret-ad-token",
        api_version="version",
    )
    assert isinstance(model_instance.openai_api_key, SecretStr)
    ...
```

Reply:
Yes. Thank you - I'd overlooked this simplification.